src/scripts/DeployToken.ts

```diff
@@ -0,0 +1,429 @@
+import {
+  encodeFunctionData,
+  encodeAbiParameters,
+  encodePacked,
+  keccak256,
+  parseUnits,
+} from 'viem';
+import { Clanker_v3_1_abi } from './Clanker';
+// Real value for the Wrapped ETH (WETH) address.
+const WETH_ADDRESS =
+  '0x4200000000000000000000000000000000000006' as `0x${string}`;
+const CLANKER_FACTORY_V3_1 =
+  '0x2A787b2362021cC3eEa3C24C4748a6cD5B687382' as `0x${string}`;
+
+// Type definitions
+interface IDeployFormData {
+  name: string;
+  symbol: string;
+  image?: File | null;
+  imageUrl?: string;
+  description?: string;
+  devBuyAmount?: string | number;
+  lockupPercentage?: number;
+  vestingUnlockDate?: bigint | null;
+  enableDevBuy?: boolean;
+  enableLockup?: boolean;
+  feeRecipient?: string | null;
+  telegramLink?: string;
+  websiteLink?: string;
+  xLink?: string;
+  marketCap?: string;
+  farcasterLink?: string;
+  pairedToken?: string;
+  creatorRewardsRecipient?: string;
+  creatorRewardsAdmin?: string;
+  creatorReward?: number;
+  interfaceAdmin?: string;
+  interfaceRewardRecipient?: string;
+}
+
+interface IClankerMetadata {
+  description?: string;
+  socialMediaUrls?: Array<{ platform: string; url: string }>;
+  auditUrls?: string[];
+}
+
+interface IClankerSocialContext {
+  interface: string;
+  platform?: string;
+  messageId?: string;
+  id?: string;
+}
+
+interface IUserInfo {
+  farcaster?: { fid?: number };
+  twitter?: { username?: string };
+}
+
+// Helper functions
+function getRelativeUnixTimestamp(unixTimestamp?: number | null): bigint {
+  if (!unixTimestamp) return BigInt(0);
+
+  let vestingDuration = BigInt(0);
+  if (unixTimestamp && BigInt(unixTimestamp) > BigInt(0)) {
+    const currentTimestamp = BigInt(Math.floor(Date.now() / 1000));
+    const targetTimestamp = BigInt(unixTimestamp);
+    if (targetTimestamp > currentTimestamp) {
+      vestingDuration = targetTimestamp - currentTimestamp;
+    } else {
+      console.warn('Target timestamp is in the past, using minimum duration');
```

Comment:
I think the script should fail instead of using minimum duration, a deployer may not want this